src/native/clr/host/typemap.cc

```diff
@@ -64,10 +64,64 @@ namespace {
 }
 
 #if defined(DEBUG)
+[[gnu::always_inline]]
+auto TypeMapper::typemap_type_to_type_debug (const char *typeName, const TypeMapEntry *map, std::string_view const& from_name, std::string_view const& to_name) noexcept -> const char*
+{
+	log_debug (LOG_ASSEMBLY, "Looking up {} type '{}'", from_name, optional_string (typeName));
+	auto equal = [](TypeMapEntry const& entry, const char *key) -> bool {
+		if (entry.from == nullptr) {
+			return 1;
+		}
+
+		return strcmp (entry.from, key) == 0;
+	};
+
+	auto less_than = [](TypeMapEntry const& entry, const char *key) -> bool {
+		if (entry.from == nullptr) {
+			return 1;
+		}
+
+		return strcmp (entry.from, key) < 0;
+	};
+
+	ssize_t idx = Search::binary_search<TypeMapEntry, const char*, equal, less_than> (typeName, map, type_map.entry_count);
+	if (idx >= 0) [[likely]] {
+		log_debug (
+			LOG_ASSEMBLY,
+			"{} type '{}' maps to {} type '{}'",
+			from_name,
+			optional_string (typeName),
+			to_name,
+			optional_string (type_map.managed_to_java[idx].to)
+		);
+		return type_map.managed_to_java[idx].to;
```

Comment:
This part looks weird; we're in a member function "type to type", with various call sites:

```
typemap_type_to_type_debug (full_type_name.get (), type_map.managed_to_java, MANAGED, JAVA);
typemap_type_to_type_debug (java_type_name, type_map.java_to_managed, JAVA, MANAGED);
```

yet in both instances we hit (*global*!) `type_map.managed_to_java`!  Shouldn't this instead be going through `map`?

Reply:
This function works correctly within the scope of this PR, the java-to-managed functionality is **not** working correctly in this PR. It is implemented in [this PR](https://github.com/dotnet/android/pull/10075) (which actually removes `typemap_type_to_type_debug`.